src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1080,6 +1080,20 @@ protected override bool IsValidPath(string path)
                 }
             }
 
+            // .NET introduced a change where invalid characters are accepted https://learn.microsoft.com/en-us/dotnet/core/compatibility/2.1#path-apis-dont-throw-an-exception-for-invalid-characters
+            // We need to check for invalid characters ourselves
+
+            // Remove drive root first
+            string pathWithoutDriveRoot = path.Substring(Path.GetPathRoot(path).Length);
+
+            foreach (string segment in pathWithoutDriveRoot.Split(Path.DirectorySeparatorChar))
```

Comment:
i suppose it may be a toss up between this and:
`pathWithoutDriveRoot.IndexOfAny(Path.GetInvalidFileNameChars().Except(Path.DirectorySeparatorChar)) != 1)`

i wondered if it might be faster to not split (but then, of course, there's the linq)

Reply:
I think in general we want to use less Linq and not more.  If there's a perf concern, it might be faster to just replace all the separators with a valid character, but at the expense of readability and only minor perf improvement I think